### PR TITLE
Resolve default fonts using fontconfig.

### LINKED ISSUE #25: All types except FamilyName::Title returning an error on Linux
```rust
use font_kit::source::SystemSource;
use font_kit::properties::*;
use font_kit::family_name::FamilyName;

SystemSource::new().select_best_match(&[FamilyName::Serif], &Properties::default()).unwrap()
```

```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: NotFound', libcore/result.rs:1009:5
```

### PR DESCRIPTION
**Pull request by RazrFalcon**
Uses the same algorithm as [fc-match](https://gitlab.freedesktop.org/fontconfig/fontconfig/blob/master/fc-match/fc-match.c).

Partially fixes #25

### CODE REVIEW

File: src/sources/fontconfig.rs [hunk 1]
```diff
@@ -103,15 +105,18 @@ impl FontconfigSource {
...
+            FcDefaultSubstitute(pattern.pattern);
+
+            let mut res = FcResultMatch;
+            let pat = FcFontMatch(ptr::null_mut(), pattern.pattern, &mut res);
+            if res != FcResultMatch {
+                return Err(SelectionError::NotFound);
+            }
 
-            let font_set = FcFontList(self.fontconfig, pattern.pattern, object_set.object_set);
+            let font_set = FcFontSetCreate();
```
Comment (RazrFalcon) [inline src/sources/fontconfig.rs]: Not sure if we should free this set. All other methods do not free it too.

**Comment by pcwalton** [comment]
Looks good!

**Comment by RazrFalcon** [comment]
Thanks. But what about Win and Mac?

**Comment by pcwalton** [comment]
I was planning to match the behavior of Web browsers: i.e. use their
defaults for serif, sans-serif, etc.

**Pull request merged**